packages/block-library/src/rss/edit.js

```diff
@@ -0,0 +1,176 @@
+/**
+ * WordPress dependencies
+ */
+import { Component, Fragment } from '@wordpress/element';
+import {
+	Button,
+	Disabled,
+	IconButton,
+	PanelBody,
+	Placeholder,
+	RangeControl,
+	ServerSideRender,
+	TextControl,
+	ToggleControl,
+	Toolbar,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import {
+	BlockControls,
+	InspectorControls,
+} from '@wordpress/editor';
+
+const DEFAULT_MIN_ITEMS = 1;
+const DEFAULT_MAX_ITEMS = 10;
+
+class RSSEdit extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.state = {
+			editing: ! this.props.attributes.feedURL,
+		};
+
+		this.toggleAttribute = this.toggleAttribute.bind( this );
+		this.onSubmitURL = this.onSubmitURL.bind( this );
+	}
+
+	toggleAttribute( propName ) {
+		return () => {
+			const value = this.props.attributes[ propName ];
+			const { setAttributes } = this.props;
+
+			setAttributes( { [ propName ]: ! value } );
+		};
+	}
+
+	onSubmitURL( event ) {
+		event.preventDefault();
+
+		const { feedURL } = this.props.attributes;
+		if ( feedURL ) {
+			this.setState( { editing: false } );
+		}
+	}
+
+	render() {
+		const {
+			blockLayout,
+			columns,
+			displayAuthor,
+			displayExcerpt,
+			displayDate,
+			excerptLength,
+			feedURL,
+			itemsToShow,
+		} = this.props.attributes;
+		const { setAttributes } = this.props;
+
+		if ( this.state.editing ) {
+			return (
+				<Placeholder
+					icon="rss"
+					label="RSS"
+				>
+					<form onSubmit={ this.onSubmitURL }>
+						<TextControl
+							placeholder={ __( 'Enter URL here…' ) }
+							value={ feedURL }
+							onChange={ ( value ) => setAttributes( { feedURL: value } ) }
+							className={ 'components-placeholder__input' }
+						/>
+						<Button
+							isLarge
+							type="submit">
+							{ __( 'Use URL' ) }
+						</Button>
+					</form>
+				</Placeholder>
+			);
+		}
+
+		const layoutControls = [
+			{
+				icon: 'list-view',
+				title: __( 'List View' ),
+				onClick: () => setAttributes( { blockLayout: 'list' } ),
+				isActive: blockLayout === 'list',
+			},
+			{
+				icon: 'grid-view',
+				title: __( 'Grid View' ),
+				onClick: () => setAttributes( { blockLayout: 'grid' } ),
+				isActive: blockLayout === 'grid',
+			},
+		];
+
+		return (
+			<Fragment>
+				<BlockControls>
+					<Toolbar>
+						<IconButton
+							className="components-icon-button components-toolbar__control"
```

Comment:
Note to self. Check in other places whether we have to use `IconButton` with all those class names listed. It might mean we could expose `ToolbarButton` component to simplify it.